components/supervisor/frontend/src/ide/ide-metrics-service-client.ts

```diff
@@ -4,31 +4,47 @@
  * See License-AGPL.txt in the project root for license information.
  */
 
-import { MetricsServicePromiseClient } from '@gitpod/ide-metrics-api-grpcweb/lib/idemetrics_grpc_web_pb'
-import { AddCounterRequest, AddCounterResponse } from '@gitpod/ide-metrics-api-grpcweb/lib/idemetrics_pb'
 import { serverUrl } from '../shared/urls';
 
-const client = new MetricsServicePromiseClient(serverUrl.asIDEMetrics().toString())
-
 export enum MetricsName {
     SupervisorFrontendClientTotal = "gitpod_supervisor_frontend_client_total",
     SupervisorFrontendErrorTotal = "gitpod_supervisor_frontend_error_total"
 }
 
+const MetricsUrl = serverUrl.asIDEMetrics().toString();
+
+interface AddCounterParam {
+    value?: number;
+    labels?: Map<string, string>;
+}
+
 export class IDEMetricsServiceClient {
 
-    static async addCounter(metricsName: string, labels?: Map<string, string>, value?: number) : Promise<AddCounterResponse> {
-        const req = new AddCounterRequest()
-        req.setName(metricsName)
+    static async addCounter(metricsName: MetricsName, labels?: Map<string, string>, value?: number) : Promise<boolean> {
+        const url = `${MetricsUrl}/metrics/counter/add/${metricsName}`
+        const params: AddCounterParam = {}
         if (value) {
-            req.setValue(value)
+            params.value = value
         }
         if (labels) {
-            const m = req.getLabelsMap()
-            for (const [name, value] of labels) {
-                m.set(name, value)
+            params.labels = labels
+        }
+        try {
+            const response = await fetch(url, {
+                method: "POST",
+                body: JSON.stringify(params),
+                headers: { "Content-Type": "application/json;" },
+                credentials: "include",
```

Comment:
why do we need it? 🤔

Reply:
I'll try to remove this line in `ide-metrics` and `credentials: "include",` in fetch, to see if it works

```
	c := cors.New(cors.Options{
		AllowCredentials: true,
```